### ISSUE TITLE
lambda positional argument wonkiness

### ISSUE THREAD

**Issue by johanatan** [borkdude/clj-kondo#249]
**version**

Latest.

**platform**

Mac.

**problem**

The following file produces two errors (with no line:column information); one for %2 and one for %3. It would seem that %1 would warrant the same warning if indeed %2 and %3 do.

**repro**

File demonstrating linter issues:
```
(ns repro)

(defn test-func []
  (let [lambda #(get-in %4 [:blah :baaaz])]
    (lambda nil nil nil {:blah {:baaaz :a-value}})))
```

Repl session illustrating correct runtime behavior:
```
cljs.user=> (defn test-func []
       #_=>   (let [lambda #(get-in %4 [:blah :baaaz])]
       #_=>     (lambda nil nil nil {:blah {:baaaz :a-value}})))
#'cljs.user/test-func
cljs.user=> (test-func)
:a-value
```

**expected behavior**

Line and column information would be included. %1 would also be warned about (or alternatively none of %1, %2 or %3 would be warned about).

**actual behavior**

Line and col info is missing. %1 is not warned about.

```
app/cljs/test.cljs::: warning: unused binding %2
app/cljs/test.cljs::: warning: unused binding %3
```

**Comment by borkdude** [comment]
Yikes, good catch. I think in case of anonymous function literal, we should not warn about unused bindings at all.

**Comment by johanatan** [comment]
I agree. I think if someone uses %N explicitly and does not use %1-%N-1 explicitly, then they are specifying that %1-%N-1 are to be ignored. It's a consequence of the fact that we are dealing with positional arguments rather than the usual ordered arguments.

**Comment by johanatan** [comment]
Btw, I hope you didn't mean to also refer to `(fn [] )` forms by "anonymous function literals". The longform anon func literal would indeed want unused binding checks (just as `defn` would)-- it's only the shortform/positional form that would not.

**Comment by borkdude** [comment]
Right. Literal refers only to `#(...)` like in https://clojure.org/guides/higher_order_functions#_function_literals.

**Comment by johanatan** [comment]
Ah, right. So, `(fn [...] ...)` is just an "anonymous function" but not a literal?

**Comment by borkdude** [comment]
Correct.

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/59bd95aa1709fdc3d6d224376a049fe6aa3268a0
